File: C# Examples/11 Building Assemblies/AddComponents/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace AddComponents
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            AddComps.m_swApp = swApp;
            using (Dialog1 dialog = new Dialog1())
            {
                dialog.ShowDialog();
            }

            return;
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 11 Building Assemblies/AddComponents - Completed/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System

Partial Class SolidWorksMacro

  Public Sub main()
    Dim MyDialog As New Dialog1
        AddComponents.m_swApp = swApp
        MyDialog.Show()
  End Sub

  ''' <summary>
  ''' The SldWorks swApp variable is pre-assigned for you.
  ''' </summary>
  Public swApp As SldWorks


End Class
